tests/parametric/test_headers_baggage.py

```diff
@@ -214,17 +214,32 @@ def test_baggageheader_maxitems_inject_D016(self, test_library):
         """Ensure that baggage headers are not injected when the number of baggage items exceeds the maximum number of items."""
         max_items = 64
         with test_library.start_span(name="test_baggageheader_maxitems_inject_D016") as span:
-            for i in range(max_items + 1):
+            for i in range(max_items + 2):
                 span.set_baggage(f"key{i}", f"value{i}")
 
             headers = test_library.inject_headers(span.span_id)
-            assert not any("baggage" in item for item in headers)
+            for header in headers:
+                if "baggage" in header:
+                    baggage_header = header
+            items = baggage_header[1].split(",")
+            assert len(items) == max_items
 
     def test_baggageheader_maxbytes_inject_D017(self, test_library):
         """Ensure that baggage headers are not injected when the total byte size of the baggage exceeds the maximum size."""
         max_bytes = 8192
         with test_library.start_span(name="test_baggageheader_maxbytes_inject_D017",) as span:
-            span.set_baggage("foo", "a" * (max_bytes))
+            baggage_items = {
+                "key1": "a" * ((max_bytes // 3)),
+                "key2": "b" * ((max_bytes // 3)),
+                "key3": "c" * ((max_bytes // 3)),
+                "key4": "d",
+            }
+            for key, value in baggage_items.items():
+                span.set_baggage(key, value)
 
-        headers = test_library.inject_headers(span.span_id)
-        assert not any("baggage" in item for item in headers)
+            headers = test_library.inject_headers(span.span_id)
+            for header in headers:
+                if "baggage" in header:
+                    baggage_header = header
+            header_size = len(baggage_header[1].encode("utf-8"))
+            assert header_size <= max_bytes
```

Comment:
Should we also assert that there are 2 items? Based on the test for max items, I think the code would be something like this:
```
items = baggage_header[1].split(",")
assert len(items) == 2
```

I think it would be only 2 items because `key1`, `key2,` and `key3` would all be about 2,735 bytes long, so `key3` would exceed the limit.
```
len(key) = 4
len("=") = 1
len(value) = 8192 // 3 = 2730
len(',') = 1

4 + 1 + 2730 + 1 = 2735
```

Reply:
Interestingly, if we dropped `key3` first (or skip it) then `key4` would fit. But the W3C spec doesn't dictate in which order to drop tags. If we drop them in order starting from the last one:
first, drop `key4` -> still too big
then drop `key3` -> now it what's left does fit

Reply:
I think we should make it clear in the RFC that we should include items in order until we reach the limit and drop everything else after that so that the tracers are more aligned. Also, the system-tests can be more specific and there would be less ambiguity.

Reply:
I'll update the RFC, thanks.